lib/browser/mqtt.ts

```diff
@@ -548,26 +620,46 @@ export class MqttClientConnection extends BufferedEventEmitter {
         if (this.currentState == MqttBrowserClientState.Connected) {
             this.currentState = MqttBrowserClientState.Stopped;
             this.emit('interrupt', -1);
+
+            /* Did we intend to disconnect? If so, then emit the event */
+            if (this.desiredState == MqttBrowserClientState.Stopped) {
+                this.emit("closed");
+            }
         }
 
         /* Only try and reconnect if our desired state is connected, ie no one has called disconnect() */
         if (this.desiredState == MqttBrowserClientState.Connected) {
             const waitTime = this.get_reconnect_time_sec();
             this.reconnectTask = setTimeout(() => {
-                    /** Emit reconnect after backoff time */
-                    this.reconnect_count++;
-                    this.connection.reconnect();
-                },
+                /** Emit reconnect after backoff time */
+                this.reconnect_count++;
```

Comment:
Preferred the previous whitespace layout tbh

Reply:
I just ran it through the default VSCode formatter. I can look at adjusting it though back to where it was 👍